engine/engine/src/engine.cpp

```diff
@@ -692,15 +692,39 @@ namespace dmEngine
 
         int write_log = dmConfigFile::GetInt(engine->m_Config, "project.write_log", 0);
         if (write_log) {
+            uint32_t count = 0;
+            char* log_paths[2];
             char sys_path[DMPATH_MAX_PATH];
-            if (dmSys::GetLogPath(sys_path, sizeof(sys_path)) == dmSys::RESULT_OK) {
-                const char* path = dmConfigFile::GetString(engine->m_Config, "project.log_dir", sys_path);
-                char full[DMPATH_MAX_PATH];
-                dmPath::Concat(path, "log.txt", full, sizeof(full));
-                dmLog::SetLogFile(full);
-            } else {
+            char main_log_path[DMPATH_MAX_PATH];
+            if (dmSys::GetLogPath(sys_path, sizeof(sys_path)) == dmSys::RESULT_OK)
+            {
+                const char* main_dir = dmConfigFile::GetString(engine->m_Config, "project.log_dir", sys_path);
```

Comment:
Doesn't this mean that we won't try to write the log to `project.log_dir` if the call to `dmSys::GetLogPath()` fails?

Should we not try three paths?

1. project.log_dir
2. dmSys::GetLogPath()
3. dmSys::GetApplicationSavePath()

Reply:
I agree, Thank you!